{
  if (this.exception.compareAndSet(null,exception) == false) {
    throw new IllegalStateException("onFailure was called multiple times");
  }
}
